### PR TITLE
domain admin restore security

### PR DESCRIPTION
**Pull request by snopoke**

### CODE REVIEW

File: corehq/apps/hqadmin/views.py [hunk 1]
```diff
@@ -467,6 +468,11 @@ def get(self, request, *args, **kwargs):
         if not self.user:
             return HttpResponseNotFound('User %s not found.' % full_username)
 
+        if 'domain' in kwargs:
+            # domain check when this get's called as domain level admin restore.
```
Comment (proteusvacuum) [inline corehq/apps/hqadmin/views.py]: Feels weird to have a check for the subclass be here. What if you made a `validate_user` method here which checks all the user properties which you could then override in the subclass?

**Review by proteusvacuum — COMMENTED** [review]
Looks fine. One non-blocking comment.

**Comment by benrudolph** [comment]
i know this wasn't you, but i feel like if we're going to have this be a route for some people it may be useful to ensure that the version param is by default set to `2.0` so that when we call `get_restore_params` it doesn't default to `1.0`. https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/ota/views.py#L175

File: corehq/apps/hqadmin/views.py (hunk 1 above)
Comment (snopoke) [inline corehq/apps/hqadmin/views.py]: dd8c6e8

**Comment by snopoke** [comment]
@benrudolph c077e4b

**Review by proteusvacuum — APPROVED** [review]

**Review by benrudolph — APPROVED** [review]

**Comment by benrudolph** [comment]
(@amstone326 is going to double check this)

**Comment by benrudolph** [comment]
ok we got the go ahead to give this a go

**Comment by benrudolph** [comment]
(will revert if it causes a bunch of issues). the play store says there are 0 installs of it, but aliza says that's usually an approximate

**Pull request merged**